src/Xamarin.Android.Build.Tasks/Utilities/NdkTools/NdkTools.cs

```diff
@@ -0,0 +1,391 @@
+
+                string GetNdkHostPlatform ()
+                {
+                        bool cannotDeterminePlatform = false;
+
+                        if (RuntimeInformation.IsOSPlatform (OSPlatform.Linux)) {
+                                if (HasPrebuiltDir (platformLinux64)) {
+                                        return platformLinux64;
+                                }
+
+                                if (HasPrebuiltDir (platformLinux32)) {
+                                        return platformLinux32;
+                                }
+
+                                cannotDeterminePlatform = true;
+                        }
+
+                        if (RuntimeInformation.IsOSPlatform (OSPlatform.OSX)) {
+                                if (HasPrebuiltDir (platformMac64)) {
+                                        return platformMac64;
+                                }
+
+                                if (HasPrebuiltDir (platformMac32)) {
+                                        return platformMac32;
+                                }
+
+                                cannotDeterminePlatform = true;
+                        }
+
+                        if (RuntimeInformation.IsOSPlatform (OSPlatform.Windows)) {
+                                if (HasPrebuiltDir (platformWindows64)) {
+                                        return platformWindows64;
+                                }
+
+                                if (HasPrebuiltDir (platformWindows32)) {
+                                        return platformWindows32;
+                                }
+
+                                cannotDeterminePlatform = true;
+                        }
+
+                        if (cannotDeterminePlatform) {
+                                throw new InvalidOperationException ($"Unable to determine host NDK platform");
+                        }
+
+                        throw new InvalidOperationException ($"Unsupported OS");
+
+                        bool HasPrebuiltDir (string name)
+                        {
+                                return Directory.Exists (Path.Combine (NdkRootDirectory, "prebuilt", name));
+                        }
+                }
+
+		static NdkVersion? ReadVersion (string androidNdkPath, TaskLoggingHelper? log = null)
+		{
+			string sourcePropertiesPath = Path.Combine (androidNdkPath, "source.properties");
+			if (!File.Exists (sourcePropertiesPath)) {
+				log?.LogCodedError ("XA5104", Properties.Resources.XA5104);
+				return null;
+			}
+
+			var splitChars = new char[] {'='};
+			string? ver = null;
+			foreach (string l in File.ReadAllLines (sourcePropertiesPath)) {
+				string line = l.Trim ();
+				if (!line.StartsWith ("Pkg.Revision", StringComparison.Ordinal)) {
+					continue;
+				}
+
+				string[] parts = line.Split (splitChars, 2);
+				if (parts.Length != 2) {
+					throw new InvalidOperationException ($"Invalid NDK version format in '{sourcePropertiesPath}'.");
```

Comment:
May be better if this is a `log?.LogDebugMessage()` + `return null`.  Should also emit an XA5104, which could be done here, or could instead be done within `Create()`; see also? https://github.com/xamarin/xamarin-android/pull/6083/files#r669778870